service/lib/dinstaller/dbus/storage/commit_actions.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+# Copyright (c) [2022] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "dbus"
+require "dinstaller/storage/commit_actions"
+
+module DInstaller
+  module DBus
+    module Storage
+      # D-Bus object to get the list of actions to perform in the system
+      class CommitActions < ::DBus::Object
+        PATH = "/org/opensuse/DInstaller/Storage/CommitActions1"
+        private_constant :PATH
+
+        INTERFACE = "org.opensuse.DInstaller.Storage.CommitActions1"
+        private_constant :INTERFACE
+
+        def initialize(logger)
+          @logger = logger
+
+          super(PATH)
+        end
+
+        dbus_interface INTERFACE do
+          dbus_method :GetAll, "out actions:as" do
+            actions_text(backend.all_actions)
+          end
+
+          dbus_method :GetMainActions, "out actions:as" do
+            actions_text(backend.main_actions)
+          end
+
+          dbus_method :GetSubvolumeActions, "out actions:as" do
+            actions_text(backend.subvolume_actions)
+          end
```

Comment:
Well, question. Why not use for that properties and having signal properties changed to catch when actions changed? Now you need to poll actions to see it changes? Or is there other way to see that actions are changed?

Reply:
Yes, this is a WIP API. I was thinking about it at night and I am going to change it. I was also thinking about a callbacks/observer mechanish. Let's try it.